src/nonlin_svm_model.jl

```diff
@@ -0,0 +1,155 @@
+export tanh_train_model, tanh_test_model#, tan_nls_model
+
+function tan_data_train()
+  #load data
+  A, b = MLDatasets.MNIST.traindata()
+  ind = findall(x -> x == 0 || x == 1, b)
+  #reshape to matrix
+  A = reshape(A,size(A,1)*size(A,2), size(A,3))./255
+
+  #get 0s and 1s
+  b = b[ind]
+  b[b.==0] .= -1
+  A = convert(Array{Float64, 2}, A[:, ind])
+  p = randperm(length(b))[1:Int(floor(length(b)/4))]
+  b = b[p]
+  A = A[:, p]
+  return A, b
+end
+
+function tan_data_test()
+  A0, b0 = MLDatasets.MNIST.testdata()
+  ind = findall(x -> x == 0 || x == 1, b)
+  A0 = reshape(A0,size(A0,1)*size(A0,2), size(A0,3))./255
+
+  #get 0s and 1s
+  b0 = b0[ind]
+  b0[b0.==0] .= -1
+  A0 = convert(Array{Float64, 2}, A0[:, ind])
+  return A0, b0
+end
+
+"""
+    model, sol = tanh_train_model(args...)
+    model, sol = tanh_test_model(args...)
+
+Return an instance of an `NLPModel` representing the hyperbolic SVM
+problem, i.e., the under-determined linear least-squares objective
+
+   f(x) = ‖1 - tanh(b ⊙ ⟨A, x⟩)‖²,
+
+where A is the data matrix with labels b = {-1, 1}ⁿ.
+
+## Arguments
+
+* `m :: Int`: the number of rows of A, size of b
+* `n :: Int`: the number of columns of A (with `n` ≥ `m`)
+
+With the MNIST Dataset, the dimensions are:
+
+    m = 12665
+    n = 784
+
+## Return Value
+
+An instance of a `FirstOrderModel` that represents the complete SVM problem in NLP form, and
+an instance of `FirstOrderNLSModel` that represents the nonlinear least squares in nonlinear least squares form.
+"""
+function svm_train_model(args...)
+  if length(args)<1
+    A, b = tan_data_train()
+  elseif length(args) == 1 || length(args) > 2
+    error("Must supply data matrix A and labels b!")
+  else
+    (A,b) = args
+  end
```

Comment:
The Julia (and non-Matlab) way to achieve this is:
```julia
svm_train_model() = svm_train_model(tan_data_train()...)

function svm_train_model(A, b)
  # ...
end
```

Reply:
Had no idea, sounds good!